File: src/main/java/com/saybetter/domain/member/application/AuthFacade.java
package com.saybetter.domain.member.application;

import org.springframework.stereotype.Component;

import com.saybetter.domain.member.application.impl.MemberService;
import com.saybetter.global.common.constant.RoleType;
import com.saybetter.global.utils.SecurityUtil;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@RequiredArgsConstructor
public class AuthFacade {

	private final SecurityUtil securityUtil;
	private final MemberService memberService;

	public void assignUserRole(RoleType role) {
		String userEmail = securityUtil.getCurrentUserEmail();
		memberService.assignUserRole(userEmail, role);
	}
}

File: core/core-api/src/main/java/io/say/better/domain/member/application/AuthFacade.kt
package io.say.better.domain.member.application;

import org.springframework.stereotype.Component;

import io.say.better.domain.member.application.impl.MemberService;
import io.say.better.storage.mysql.domain.entity.Member;
import io.say.better.core.enums.RoleType;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Component
@RequiredArgsConstructor
public class AuthFacade {

	private final MemberService memberService;

	public void assignUserRole(RoleType role) {
		Member member = memberService.getCurrentMember();
		memberService.assignUserRole(member.getEmail(), role);
	}
}
